src/Components/Components/src/Routing/TemplateParser.cs

```diff
@@ -13,21 +13,20 @@ namespace Microsoft.AspNetCore.Components.Routing
     // simple parameters from it.
     // Some differences with ASP.NET Core routes are:
     // * We don't support catch all parameter segments.
-    // * We don't support optional parameter segments.
     // * We don't support complex segments.
     // The things that we support are:
     // * Literal path segments. (Like /Path/To/Some/Page)
     // * Parameter path segments (Like /Customer/{Id}/Orders/{OrderId})
     internal class TemplateParser
     {
         public static readonly char[] InvalidParameterNameCharacters =
-            new char[] { '*', '?', '{', '}', '=', '.' };
+            new char[] { '*', '{', '}', '=', '.' };
 
         internal static RouteTemplate ParseTemplate(string template)
         {
             var originalTemplate = template;
             template = template.Trim('/');
-            if (template == "")
+            if (template == String.Empty)
```

Comment:
We prefer using the C# language keyword over the .NET type: https://github.com/dotnet/aspnetcore/wiki/Engineering-guidelines#use-c-type-keywords-in-favor-of-net-type-names. So this would be

Reply:
```suggestion
            if (template == string.Empty)
```

Reply:
That's a good reference. Fixed.